framework/libra-framework/sources/ol_sources/vote_lib/multi_action.move

```diff
@@ -173,19 +199,293 @@ module ol_framework::multi_action {
         vote: ballot::new_tracker<Proposal<PropGovSigners>>(),
       });
     };
+
+    if (!exists<Offer>(multisig_address)) {
+      move_to(sig, construct_empty_offer());
+    };
+  }
+
+  // TODO: remove this after offer migration is completed
+  #[test_only]
+  public(friend) fun init_gov_deprecated(sig: &signer) {
+    let multisig_address = signer::address_of(sig);
+
+    if (!exists<Governance>(multisig_address)) {
+        move_to(sig, Governance {
+        cfg_duration_epochs: DEFAULT_EPOCHS_EXPIRE,
+        cfg_default_n_sigs: 0, // deprecate
+        signers: vector::empty(),
+        withdraw_capability: option::none(),
+        guid_capability: account::create_guid_capability(sig),
+      });
+    };
+
+    if (!exists<Action<PropGovSigners>>(multisig_address)) {
+      move_to(sig, Action<PropGovSigners> {
+        can_withdraw: false,
+        vote: ballot::new_tracker<Proposal<PropGovSigners>>(),
+      });
+    };
+  }
+
+
+  // Private function to assist governance vote
+  fun add_offer_addresses(addr: address, proposed: vector<address>) acquires Offer {
+    let offer = borrow_global_mut<Offer>(addr);
+    let duration = epoch_helper::get_current_epoch() + DEFAULT_EPOCHS_OFFER_EXPIRE;
+    let i = 0;
+    while (i < vector::length(&proposed)) {
+      let addr = vector::borrow(&proposed, i);
+      vector::push_back(&mut offer.proposed, *addr);
+      vector::push_back(&mut offer.expiration_epoch, duration);
+      i = i + 1;
+    };
+  }
+
+  // DANGER - may forge the signer of the multisig account is necessary here
+  // TODO: remove this function after offer migration is completed
+  // Migrate a legacy account to have structure Offer in order to propose authorities changes
+  public entry fun migrate_offer(sig: &signer, multisig_address: address) {
```

Comment:
I agree this seems to be the only possible migration pattern. I think this feature could be gated with Feature flags, or has an epoch expiration.
In reality it only seems 3 accounts currently would need to use it.

Last, you might consider moving the migration to a dedicated module so that you can create another layer of security.

For practical purposes we also need a PR that removes this feature at a later version.

Reply:
We have 6 accounts multisign at the moment and we might have some with governance already initiated.

![image](https://github.com/0LNetworkCommunity/libra-framework/assets/10797037/f228a9b7-b550-47d6-8aba-f8994bbc1650)